File: Assets/Bundles/Shared/Resources/TS/UserInput/Drivers/TouchscreenDriver.lua
-- Compiled with unity-ts v2.1.0-75
local Signal = require("Shared/TS/Util/Signal").Signal
local TouchscreenDriver
do
	TouchscreenDriver = setmetatable({}, {
		__tostring = function()
			return "TouchscreenDriver"
		end,
	})
	TouchscreenDriver.__index = TouchscreenDriver
	function TouchscreenDriver.new(...)
		local self = setmetatable({}, TouchscreenDriver)
		return self:constructor(...) or self
	end
	function TouchscreenDriver:constructor()
		self.Touch = Signal.new()
		self.TouchTap = Signal.new()
		UserInputService.InputProxy:OnTouchEvent(function(touchIndex, position, phase)
			self.Touch:Fire(touchIndex, position, phase)
		end)
		UserInputService.InputProxy:OnTouchTapEvent(function(touchIndex, position, phase)
			self.TouchTap:Fire(touchIndex, position, phase)
		end)
	end
	function TouchscreenDriver:instance()
		if self.inst == nil then
			self.inst = TouchscreenDriver.new()
		end
		return self.inst
	end
end
return {
	TouchscreenDriver = TouchscreenDriver,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Shared/UserInput/Drivers/TouchscreenDriver.ts
import { Signal } from "Shared/Util/Signal";

export class TouchscreenDriver {
	private static inst: TouchscreenDriver;

	public readonly touch = new Signal<[touchIndex: number, position: Vector3, phase: TouchPhase]>();
	public readonly touchTap = new Signal<[touchIndex: number, position: Vector3, phase: InputActionPhase]>();

	private constructor() {
		// InputBridge.Instance.OnTouchEvent((touchIndex, position, phase) => {
		// 	this.Touch.Fire(touchIndex, position, phase);
		// });
		// InputBridge.Instance.OnTouchTapEvent((touchIndex, position, phase) => {
		// 	this.TouchTap.Fire(touchIndex, position, phase);
		// });
	}

	/** **NOTE:** Internal only. Use `Touchscreen` class instead. */
	public static Instance() {
		return (this.inst ??= new TouchscreenDriver());
	}
}
